groth16-framework/test_data/Groth16VerifierExtensions.sol

```diff
@@ -208,6 +253,26 @@ contract Query2 is Verifier {
         return result;
     }
 
+    // Convert a bytes32 to an uint256.
+    function convertBytes32ToU256(bytes32 b) internal pure returns (uint256) {
+        uint256 result;
+        for (uint32 i = 0; i < 32; i++) {
+            result |= uint256(uint8(b[i])) << (8 * i);
+        }
+
+        return result;
+    }
+
+    // Convert the specified byte slice to an uint256.
+    function convertByteSliceToU256(bytes memory pis, uint32 offset) internal pure returns (uint256) {
+        uint256 result;
+        for (uint32 i = 0; i < 8 ; ++i) {
+            result |= uint256(convertToU32(pis, offset + i * 8));
```

Comment:
Isn't a shift missing here?

Reply:
Sorry, which line do you mean?

Reply:
Line 270: shouldn't it be `result |= uint256(convertToU32(pis, offset + i * 8)) << (32*i);`? Because we should add 32-bit limbs in little-endian order, but here it seems to me we are always add a u32 to the least significant bits of `result`. Am I missing something?

Reply:
I see, thank you. I fixed in commit [4a1defd](https://github.com/Lagrange-Labs/Euclid-database/pull/21/commits/4a1defd2fcaefe464b1992278208d1fd3e81fbed) with updated test values. I am running the test with the above query `identifier` fix.